umbraco/Umbraco-CMS
# V8: Make sure media picker items are sortable

### Prerequisites


If there's an existing issue for this PR then this fixes https://github.com/umbraco/Umbraco-CMS/issues/6035

### Description

Items picked in a media picker with "Pick multiple items" enabled should be sortable, but they are not. See #6035 for more details.

This PR fixes the sortable implementation of the media picker. When it is applied, the picked items can be sorted:

![media-picker-sortable-after](https://user-images.githubusercontent.com/7405322/62286117-63bd1000-b457-11e9-8fcd-0b04caaac14b.gif)

Note that I have removed the `$scope.$watch` as part of this PR as it didn't seem to have any positive effect. I removed it because it caused a reload of all images when sorting the media picker items, thus executing a bunch of unnecessary requests.

#### Testing this PR

1. Verify that media picker items _cannot_ be sorted when "Pick multiple items" is _disabled_ on the media picker config.
2. Verify that media picker items _can_ be sorted when "Pick multiple items" is _enabled_ on the media picker config.
3. Verify that the updated item sort order is persisted when saving the content.
